app/models/katello/concerns/hostgroup_extensions.rb

```diff
@@ -88,14 +88,19 @@ def content_and_puppet_match?
       end
 
       def equivalent_kickstart_repository
-        return unless operatingsystem && kickstart_repository
+        return unless operatingsystem &&
+                      kickstart_repository &&
+                      operatingsystem.class.method_defined?(:kickstart_repos)
```

Comment:
I think it's more appropriate to use `respond_to?` vs `method_defined?`

my thinking is 'cause:
- why use a Rails API when we can use a Ruby API
- `respond_to?` more accurately describes what we want to know ie does the object respond to some message, not whether or not it implements the code in order to respond to some message (if I understand the difference b/t the two properly)

Reply:
Thanks, that is the method I was trying to think of :) Let me try it out